src/vecdb/vecdb.rs

```diff
@@ -28,22 +84,32 @@ impl VecDb {
         splitter_window_size: usize,
         splitter_soft_limit: usize,
         embedding_model_name: String,
+        embedding_endpoint: String,
+        embedding_endpoint_style: String,
     ) -> Result<VecDb, String> {
         let handler = match VecDBHandler::init(cache_dir, embedding_size).await {
             Ok(res) => res,
             Err(err) => { return Err(err) }
         };
-        let vecdb_handler = Arc::new(Mutex::new(handler));
-        let retriever_service = Arc::new(Mutex::new(FileVectorizerService::new(
-            vecdb_handler.clone(), cooldown_secs, splitter_window_size, splitter_soft_limit,
-            embedding_model_name.clone(), cmdline.api_key.clone(),
+        let vecdb_handler = Arc::new(AMutex::new(handler));
+        let retriever_service = Arc::new(AMutex::new(FileVectorizerService::new(
+            vecdb_handler.clone(),
+            cooldown_secs,
+            splitter_window_size,
+            splitter_soft_limit,
+            embedding_model_name.clone(),
+            cmdline.api_key.clone(),
```

Comment:
Let's rethink using of the internal api_key 
Maybe we should add extra args like `hf_api_key`, `openai_api_key` and use them instead

Reply:
@olegklimov do you think we should?